nifi-nar-bundles/nifi-riemann-bundle/nifi-riemann-reporting-task/src/main/java/org/apache/nifi/reporting/riemann/RiemannReportingTask.java

```diff
@@ -0,0 +1,244 @@
+            .allowableValues("true", "false").required(true).defaultValue("true").addValidator(StandardValidators.BOOLEAN_VALIDATOR).build();
+    static final PropertyDescriptor MIN_BULLETIN_LEVEL = new PropertyDescriptor.Builder().name("Minimum Bulletin Level")
+            .description("Only forward bulletin messages at this level and above to Riemann").allowableValues(LogLevel.values()).required(true).defaultValue("WARNING").build();
+    protected volatile Transport transport;
+    private volatile long lastObservedBulletinId = 0;
+    private volatile RiemannClient riemannClient = null;
+    private volatile MetricsService metricsService;
+    private volatile VirtualMachineMetrics virtualMachineMetrics = VirtualMachineMetrics.getInstance();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = Lists.newArrayList();
+        properties.add(RIEMANN_HOST);
+        properties.add(RIEMANN_PORT);
+        properties.add(TRANSPORT_PROTOCOL);
+        properties.add(SERVICE_PREFIX);
+        properties.add(HOSTNAME);
+        properties.add(TAGS);
+        properties.add(SEND_JVM_METRICS);
+        properties.add(SEND_NIFI_METRICS);
+        properties.add(SEND_PROCESSOR_METRICS);
+        properties.add(SEND_BULLETIN_MESSAGES);
+        properties.add(MIN_BULLETIN_LEVEL);
+        properties.add(WRITE_TIMEOUT);
+        return properties;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder().name(propertyDescriptorName).required(false).addValidator(StandardValidators.NON_EMPTY_VALIDATOR).expressionLanguageSupported(false).dynamic(true)
+                .build();
+    }
+
+    @OnStopped
+    public final void cleanUpClient() {
+        if (riemannClient != null) {
+            this.riemannClient.close();
+        }
+        this.riemannClient = null;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ConfigurationContext context) throws ProcessException {
+        if (riemannClient == null || !riemannClient.isConnected()) {
+            transport = Transport.valueOf(context.getProperty(TRANSPORT_PROTOCOL).getValue());
+            String host = context.getProperty(RIEMANN_HOST).evaluateAttributeExpressions().getValue();
+            int port = context.getProperty(RIEMANN_PORT).asInteger();
+            RiemannClient client = null;
+            try {
+                switch (transport) {
+                case TCP:
+                    client = RiemannClient.tcp(host, port);
+                    break;
+                case UDP:
+                    client = RiemannClient.udp(host, port);
+                    break;
+                }
```

Comment:
Since the `TRANSPORT_PROTOCOL` property only allows the `Transport.TCP` and `Transport.UDP` values, we shouldn't have to worry about handling an additional value.